tests/test_client_proto.py

```diff
@@ -252,3 +252,22 @@ async def test_connection_lost_sets_transport_to_none(
     proto.connection_lost(OSError())
 
     assert proto.transport is None
+
+
+async def test_connection_lost_exception_is_marked_retrieved(
+    loop: asyncio.AbstractEventLoop,
+) -> None:
+    """Test that connection_lost properly handles exceptions without warnings."""
+    proto = ResponseHandler(loop=loop)
+    proto.connection_made(mock.Mock())
+
+    # Simulate an SSL shutdown timeout error
+    ssl_error = TimeoutError("SSL shutdown timed out")
+    proto.connection_lost(ssl_error)
+
+    # Verify the exception was set on the closed future
+    assert proto.closed.done()
+    exc = proto.closed.exception()
+    assert exc is not None
+    assert "Connection lost: SSL shutdown timed out" in str(exc)
```

Comment:
This isn't great to test this but otherwise we have to dig into the future internals which didn't seem like a good idea.